test/functional/provider/clipboard_spec.lua

```diff
@@ -752,4 +752,13 @@ describe('clipboard (with fake clipboard.vim)', function()
     expect('some some')
     eq('some', eval('getreg("*")'))
   end)
+
+  it('has correct value for clipboard registers #24257', function ()
+    -- disable clipboard provider to test this behavior on non-X11 systems
+    command('let g:clipboard = ""')
+    eval('setreg("*", "")')
+    eval('setreg("+", "waddup")')
+    eq('', eval('getreg("*")'))
+    eq('waddup', eval('getreg("+")'))
+  end)
```

Comment:
The test is wrong and has nothing to do the the issue.
```suggestion
  it('does not fall back to unnamed register with getreg() #24257', function ()
    eval('setreg("", "wrong")')
    command('let g:cliperror = 1')
    eq('', eval('getreg("*")'))
    eq('', eval('getreg("+")'))
  end)
```

Reply:
This is definitely better more descriptive, since falling back to unnamed is the behavior we don't have to happen. My test accomplished the same thing because the second `setreg` sets the unnamed register to "waddup", but yours is more direct and readable 👍

Reply:
But you didn't set the unnamed register. You set the `+` register.

Reply:
oh interesting, I thought the unnamed register would always be set as a result of setreg, but maybe that is not the case.